In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import datetime as dt


In [2]:
customer=pd.read_csv("CUSTOMERS.CSV")
geo_location=pd.read_csv("GEO_LOCATION.csv")
order_items = pd.read_csv("ORDER_ITEMS.csv")
order_payment=pd.read_csv("ORDER_PAYMENTS.csv")
order_review=pd.read_csv("ORDER_REVIEW_RATINGS.csv")
orders=pd.read_csv("ORDERS.csv")
products=pd.read_csv("PRODUCTS.csv")
sellers=pd.read_csv("SELLERS.csv")

In [3]:
customer.rename(columns={"customer_zip_code_prefix": "zip_code"},inplace=True)
customer.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   customer_id         99441 non-null  object
 1   customer_unique_id  99441 non-null  object
 2   zip_code            99441 non-null  int64 
 3   customer_city       99441 non-null  object
 4   customer_state      99441 non-null  object
dtypes: int64(1), object(4)
memory usage: 3.8+ MB


In [4]:
geo_location.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19015 entries, 0 to 19014
Data columns (total 5 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   geolocation_zip_code_prefix  19015 non-null  int64  
 1   geolocation_lat              19015 non-null  float64
 2   geolocation_lng              19015 non-null  float64
 3   geolocation_city             19015 non-null  object 
 4   geolocation_state            19015 non-null  object 
dtypes: float64(2), int64(1), object(2)
memory usage: 742.9+ KB


In [5]:
geo_location.rename(columns={"geolocation_zip_code_prefix":"zip_code"},inplace=True)


In [6]:
df=pd.merge(left=customer,right=geo_location,on="zip_code",how="inner")

In [7]:
df1=pd.merge(left=order_items,right=order_payment,on="order_id",how="inner")
df1=pd.merge(left=df1,right=order_review,on="order_id",how="inner")
df1=pd.merge(left=df1,right=orders,on="order_id",how="inner")
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 118315 entries, 0 to 118314
Data columns (total 22 columns):
 #   Column                         Non-Null Count   Dtype  
---  ------                         --------------   -----  
 0   order_id                       118315 non-null  object 
 1   order_item_id                  118315 non-null  int64  
 2   product_id                     118315 non-null  object 
 3   seller_id                      118315 non-null  object 
 4   shipping_limit_date            118315 non-null  object 
 5   price                          118315 non-null  float64
 6   freight_value                  118315 non-null  float64
 7   payment_sequential             118315 non-null  int64  
 8   payment_type                   118315 non-null  object 
 9   payment_installments           118315 non-null  int64  
 10  payment_value                  118315 non-null  float64
 11  review_id                      118315 non-null  object 
 12  review_score                  

In [8]:
df1.isnull().sum()

order_id                            0
order_item_id                       0
product_id                          0
seller_id                           0
shipping_limit_date                 0
price                               0
freight_value                       0
payment_sequential                  0
payment_type                        0
payment_installments                0
payment_value                       0
review_id                           0
review_score                        0
review_creation_date                0
review_answer_timestamp             0
customer_id                         0
order_status                        0
order_purchase_timestamp            0
order_approved_at                  15
order_delivered_carrier_date     1254
order_delivered_customer_date    2588
order_estimated_delivery_date       0
dtype: int64

### null values
order approved at = 15 
order_delivered_carrier_date = 1254
order_delivered_customer_date = 2588


In [9]:
(df1.isnull().sum()*100)/len(df1)

order_id                         0.000000
order_item_id                    0.000000
product_id                       0.000000
seller_id                        0.000000
shipping_limit_date              0.000000
price                            0.000000
freight_value                    0.000000
payment_sequential               0.000000
payment_type                     0.000000
payment_installments             0.000000
payment_value                    0.000000
review_id                        0.000000
review_score                     0.000000
review_creation_date             0.000000
review_answer_timestamp          0.000000
customer_id                      0.000000
order_status                     0.000000
order_purchase_timestamp         0.000000
order_approved_at                0.012678
order_delivered_carrier_date     1.059883
order_delivered_customer_date    2.187381
order_estimated_delivery_date    0.000000
dtype: float64

In [16]:
df1 = df1.dropna()

In [17]:
df1.isnull().sum()

order_id                         0
order_item_id                    0
product_id                       0
seller_id                        0
shipping_limit_date              0
price                            0
freight_value                    0
payment_sequential               0
payment_type                     0
payment_installments             0
payment_value                    0
review_id                        0
review_score                     0
review_creation_date             0
review_answer_timestamp          0
customer_id                      0
order_status                     0
order_purchase_timestamp         0
order_approved_at                0
order_delivered_carrier_date     0
order_delivered_customer_date    0
order_estimated_delivery_date    0
dtype: int64

In [18]:
df.isnull().sum()

customer_id           0
customer_unique_id    0
zip_code              0
customer_city         0
customer_state        0
geolocation_lat       0
geolocation_lng       0
geolocation_city      0
geolocation_state     0
dtype: int64

In [25]:
total_revenue=df1.payment_value.sum()
print("TOTAL REVENUE = ",total_revenue)

TOTAL REVENUE =  19880624.94


In [27]:
df1.head()

,order_id,order_item_id,product_id,seller_id,shipping_limit_date,price,freight_value,payment_sequential,payment_type,payment_installments,...,review_score,review_creation_date,review_answer_timestamp,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date
0,00010242fe8c5a6d1ba2dd792cb16214,1,4244733e06e7ecb4970a6e2683c13e61,48436dade18ac8b2bce089ec2a041202,9/19/2017 9:45,58.90,13.29,1,credit_card,2,...,5,9/21/2017 0:00,9/22/2017 10:57,3ce436f183e68e07877b285a838db11a,delivered,9/13/2017 8:59,9/13/2017 9:45,9/19/2017 18:34,9/20/2017 23:43,9/29/2017 0:00
1,00018f77f2f0320c557190d7a144bdd3,1,e5f2d52b802189ee658865ca93d83a8f,dd7ddc04e1b6c2c614352b383efe2d36,5/3/2017 11:05,239.90,19.93,1,credit_card,3,...,4,5/13/2017 0:00,5/15/2017 11:34,f6dd3ec061db4e3987629fe6b26e5cce,delivered,4/26/2017 10:53,4/26/2017 11:05,5/4/2017 14:35,5/12/2017 16:04,5/15/2017 0:00
2,000229ec398224ef6ca0657da4fc703e,1,c777355d18b72b67abbeef9df44fd0fd,5b51032eddd242adc84c38acab88f23d,1/18/2018 14:48,199.00,17.87,1,credit_card,5,...,5,1/23/2018 0:00,1/23/2018 16:06,6489ae5e4333f3693df5ad4372dab6d3,delivered,1/14/2018 14:33,1/14/2018 14:48,1/16/2018 12:36,1/22/2018 13:19,2/5/2018 0:00
3,00024acbcdf0a6daa1e931b038114c75,1,7634da152a4610f1595efa32f14722fc,9d7a1d34a5052409006425275ba1c2b4,8/15/2018 10:10,12.99,12.79,1,credit_card,2,...,4,8/15/2018 0:00,8/15/2018 16:39,d4eb9395c8c0431ee92fce09860c5a06,delivered,8/8/2018 10:00,8/8/2018 10:10,8/10/2018 13:28,8/14/2018 13:32,8/20/2018 0:00
4,00042b26cf59d7ce69dfabb4e55b4fd9,1,ac6c3623068f30de03045865e4e10089,df560393f3a51e74553ab94004ba5c87,2/13/2017 13:57,199.90,18.14,1,credit_card,3,...,5,3/2/2017 0:00,3/3/2017 10:54,58dbd0b2d70206bf40e62cd34e84d795,delivered,2/4/2017 13:57,2/4/2017 14:10,2/16/2017 9:46,3/1/2017 16:42,3/17/2017 0:00
